### ISSUE TITLE
Redirect to Stripe is failing

### ISSUE THREAD

**Issue by jsandrae** [umbraco/Umbraco.Commerce.PaymentProviders.Stripe#9]
Submitting the form on the checkout review page is causing some users to proceed to the Order Confirmation page without being redirected to Stripe for payment processing. We've been unable to recreate the issues ourselves, but we've been getting reports of it happening a few times out of every 100 orders (~1-2 times per day). We've reached out to Umbraco Commerce support and they recommended we submit an issue here. 

We have been able to recreate the issue with JavaScript disabled, but JavaScript is required earlier in the checkout process and therefore no users should be able to reach the Order Review page without JS. It can also happen if `window.handleStripeCheckout(event)` throws an error on the Review Page form, though we're not sure what could cause this. Since the redirect occurs client-side, it's proving difficult to look into the issue. 

Are you aware of any situations where the Stripe redirect script has an error outside of JavaScript being disabled? Do you have any guidance in how we can identify the issue?

---
_This item has been added to our backlog AB#47196_

**Comment by mattbrailsford** [comment]
Ok, so I think there are two things we can do here:

1) Improve the javascript that is generated here https://github.com/umbraco/Umbraco.Commerce.PaymentProviders.Stripe/blob/main/src/Umbraco.Commerce.PaymentProviders.Stripe/StripeCheckoutPaymentProvider.cs#L239 and ensure any errors are presented to the user and ensure the form doesn't auto submit

and 2) maybe it makes more sense the that form generated here https://github.com/umbraco/Umbraco.Commerce.PaymentProviders.Stripe/blob/main/src/Umbraco.Commerce.PaymentProviders.Stripe/StripeCheckoutPaymentProvider.cs#L227 defaults to the ErrorUrl rather than the ContinueUrl. You should never really get to that page as the JS should catch the submit and do it's thing, but it appears that's not always the case so maybe in those rare occassions, it's better to submit to the error page.

**Comment by umbracotrd** [comment]
@jsandrae just to clarify, you're using Umbraco 13.5.1, Commerce 3.1.12, and Stripe Checkout 13.1.0, right?

**Comment by jsandrae** [comment]
Correct: Umbraco version 13.5.1, Umbraco.Commerce Version: 13.1.12, Umbraco.Commerce.PaymentProviders.Stripe 13.1.0

**Comment by umbracotrd** [comment]
I've changed two things follow @mattbrailsford 's suggestions:
- I migrated away from the stripe js library because `stripe.redirectToCheckout` [is deprecated](https://docs.stripe.com/js/deprecated/redirect_to_checkout). The new js will redirect buyer using the good old `window.location.href`.
- I also set the form action to the payment provider's error url instead of continue url. With that change, if the form is somehow submitted, god forbids that, the buyer will be redirected to the error url instead of the continue (success) url like the current version.

The change details can be see in #10.

I pushed version `13.1.1--preview.2+4576af2` to [Umbraco nightly nuget feeds](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/install/installing-nightly-builds) in case anyone wants to try out the new version early. If there're no complains during the weekend, I'll release version `13.1.1` to nuget on Monday.
